### ISSUE TITLE
Test data may be leaking into the training data.

### ISSUE THREAD

**Issue by imenurok** [andrew-jeremy/Reducing-Deep-Network-Complexity-with-Fourier-Transform-Methods#2]
Hi!
I am interested in data argumentation on images and I read your interesting paper.
Your experimental results look amazing, but I had a small doubt about your implementation.

> _, test_loss = sess.run([optimizer, cost], feed_dict={x: test_x, y: y_test, keep_prob: 1.0})

I am afraid that your implementation will learn not only training data but also test data by the above line.
Actually, if some lines which have something to do with the above line was changed on CIFAR-100 training, I got bad result as below.

> ('Epoch:', '0300', 'training cost=', '0.199868858', 'training acc=', '0.97720', 'val acc=', '0.06470')

It suggests that your results will have been incorrect by the data leakage, so your method will not have been compared with other methods correctly. 
I recommend checking your implementation ASAP!

**Comment by spotofleopard** [comment]
I am so disappointed the author didn't get 101% accuracy, with such _advanced_ technique of "FFT".
BTW: I ran the mnist script, it only gets 97.76% accuracy, nowhere near his 99.99% claim.

**Comment by tgeorgy** [comment]
Hi, one more point - FFT is a linear transformation. Applying it right before fully-connected layer should not make any difference.

**Comment by brunner9001** [comment]
Ironically, the MNIST script doesn't seem to contain the critical flaw. But then again it doesn't offer good performance either. ;)

I suspect he had originally used the same flawed method for MNIST (hence 99.99 in the paper), and then modified it to use Keras.
